### ISSUE TITLE
Eager relationships not loaded when using preload

### ISSUE THREAD

**Issue by gerardolm** [typeorm/typeorm#8944]
### Expected Behavior

When defining a one-to-many relationship with `eager: true` in the "many" property within the "one" side (in other words, many-side eager loading) and using `preload` method on "one" entity, "many" side is expected to be loaded eagerly as specified by documentation:

> Creates a new entity from the given plain javascript object. If the entity already exists in the database, then it loads it (**and everything related to it**), replaces all values with the new ones from the given object, and returns the new entity. The new entity is actually an entity loaded from the database with all properties replaced from the new object.

### Actual Behavior

Many side of the relationship is not loaded (left undefined).

If this behaviour is expected, then it should be detailed in documentation, as the "everything related to it" block leads the reader to think eager relationships are loaded. In other words, either code is wrong or doc is wrong.

### Steps to Reproduce

1. Define two entities with a one-to-many relationship between them.
2. Use preload on the "one" side.
3. See that "many" side is not loaded.

Truncated code for the entities:

```typescript
@Entity()
export class A {
  @OneToMany(() => B, (b) => b.a, {
    cascade: ['soft-remove'],
    eager: true,
  })
  bs!: B[];
}
```

```typescript
@Entity()
export class B {
  @JoinColumn({ name: 'aId' })
  @ManyToOne(() => A, { nullable: false })
  a!: A;
}
```

### My Environment

| Dependency          | Version  |
| ---                 | ---      |
| Operating System    | WSL Ubuntu on Windows 10 |
| Node.js version     | 18.0.0  |
| Typescript version  | 4.6.3  | 
| TypeORM version     | 0.3.5  | 

### Relevant Database Driver(s)

| DB Type           | Reproducible  |
|-------------------| ---           |
| `aurora-mysql`    | no            |
| `aurora-postgres` | no            |
| `better-sqlite3`  | no            |
| `cockroachdb`     | no            |
| `cordova`         | no            |
| `expo`            | no            |
| `mongodb`         | no            |
| `mysql`           | no            |
| `nativescript`    | no            |
| `oracle`          | no            |
| `postgres`        | yes            |
| `react-native`    | no            |
| `sap`             | no            |
| `spanner`         | no            |
| `sqlite`          | no            |
| `sqlite-abstract` | no            |
| `sqljs`           | no            |
| `sqlserver`       | no            |

### Are you willing to resolve this issue by submitting a Pull Request?

 - ✖️ Yes, I have the time, and I know how to start.
 - ✖️ Yes, I have the time, but I don't know how to start. I would need guidance.
 - ✖️ No, I don’t have the time, but I can support (using donations) development.
 - ✅ No, I don’t have the time and I’m okay to wait for the community / maintainers to resolve this issue.

**Comment by spuxx1701** [comment]
Can confirm this behavior also for OneToOne and ManyToMany relations.

**Comment by jaypea** [comment]
this also applies if Entity B is preloaded without eager setting. the related entity A is not fetched.

**Comment by deki23** [comment]
Is there any workaround for this?

**Comment by Picoquio** [comment]
More than two years, please.

**Comment by Dmtreaqq** [comment]
Maybe we just don't understand something, any updates on it, please ?

**Comment by Yokutto** [comment]
I think this one needs to stay opened

**Comment by Cprakhar** [comment]
> this also applies if Entity B is preloaded without eager setting. the related entity A is not fetched.

@jaypea, you are correct. According to the documentation, the preload method should return everything related to the entity.
But first need to confirm with the project maintainers before jumping into the fix.

**Comment by Cprakhar** [comment]
> > this also applies if Entity B is preloaded without eager setting. the related entity A is not fetched.
> 
> [@jaypea](https://github.com/jaypea), you are correct. According to the documentation, the preload method should return everything related to the entity. But first need to confirm with the project maintainers before jumping into the fix.

@gioboa, what should the `preload` method's correct behavior be? I want to take this up.
